### PR TITLE
215 return last entries

### PR DESCRIPTION
**Pull request by josepato87**
Added the endpoint for returning last entries, besides has a refactor of "TimeEntries". 

There are some format changes, please take a look and let us know if we need to change them again.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from time_tracker_api import create_app
 from time_tracker_api.database import init_sql
 from time_tracker_api.security import get_or_generate_dev_secret_key
-from time_tracker_api.time_entries.time_entries_model import (
```
Comment (scastillo-jp) [inline tests/conftest.py:14]: @josepato87 if it is an unused line, I recommend removing the commented lines

**Review by scastillo-jp — CHANGES_REQUESTED** [review]

File: time_tracker_api/time_entries/time_entries_namespace.py [hunk 2]
```diff
@@ -256,9 +256,28 @@ def post(self):
         return time_entries_dao.create(ns.payload), HTTPStatus.CREATED
 
 
+# TODO: Once this endpoint is working as expected in prod, review and
+# remove unnecessary filter args. As we are using same attributes_filter
```
Comment (Angeluz-07) [inline time_tracker_api/time_entries/time_entries_namespace.py:260]: @PaulRC-ioet  Let's remove these lines and use a new attributes_filter for this endpoint.

File: time_tracker_api/time_entries/time_entries_namespace.py [hunk 3]
```diff
@@ -256,9 +256,28 @@ def post(self):
...
+        conditions = attributes_filter.parse_args()
+
+        return time_entries_dao.get_lastest_entries_by_project(
+            conditions=conditions
+        )
+
+
 @ns.route('/<string:id>')
 @ns.response(HTTPStatus.NOT_FOUND, 'This time entry does not exist')
-@ns.response(HTTPStatus.UNPROCESSABLE_ENTITY, 'The id has an invalid format')
```
Comment (Angeluz-07) [inline time_tracker_api/time_entries/time_entries_namespace.py:261]: I have the doubt of the reason to remove this line.

File: tests/conftest.py (hunk 1 above)
Comment (PaulRC-ioet) [reply tests/conftest.py:14]: As we discuss with @Angeluz-07 we refactor the code because time_entries_model had multiple responsibilities.

**Review by Angeluz-07 — CHANGES_REQUESTED** [review]
We are almost done. Only a few fixes.

**Review by scastillo-jp — APPROVED** [review]

File: time_tracker_api/time_entries/time_entries_namespace.py (hunk 2 above)
Comment (PaulRC-ioet) [reply time_tracker_api/time_entries/time_entries_namespace.py:260]: the attribute filters were removed and the TODO line deleted

File: time_tracker_api/time_entries/time_entries_namespace.py (hunk 3 above)
Comment (PaulRC-ioet) [reply time_tracker_api/time_entries/time_entries_namespace.py:261]: this line was added again

**Review by Angeluz-07 — APPROVED** [review]

File: time_tracker_api/time_entries/time_entries_dao.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
+import abc
+from commons.data_access_layer.cosmos_db import (
+    CosmosDBDao,
+    CustomError,
+)
+from utils.extend_model import (
+    add_project_info_to_time_entries,
+    add_activity_name_to_time_entries,
+    create_custom_query_from_str,
+    add_user_email_to_time_entries,
```
Comment (enriquezrene) [inline time_tracker_api/time_entries/time_entries_dao.py:10]: This function is not used so you can remove it
```
add_user_email_to_time_entries
```

File: time_tracker_api/time_entries/time_entries_namespace.py [hunk 5]
```diff
@@ -256,6 +256,17 @@ def post(self):
         return time_entries_dao.create(ns.payload), HTTPStatus.CREATED
 
 
+@ns.route('/latest')
+class LatestTimeEntries(Resource):
+    @ns.doc('list_latest_time_entries')
+    @ns.marshal_list_with(time_entry)
+    @ns.response(HTTPStatus.NOT_FOUND, 'Time entry not found')
```
Comment (enriquezrene) [inline time_tracker_api/time_entries/time_entries_namespace.py:263]: A better message would be:
```
No time entries found.
```
It is not returning one by many, right?

**Review by enriquezrene — CHANGES_REQUESTED** [review]

File: time_tracker_api/time_entries/time_entries_namespace.py (hunk 5 above)
Comment (PaulRC-ioet) [reply time_tracker_api/time_entries/time_entries_namespace.py]: The error message is changed already

File: time_tracker_api/time_entries/time_entries_dao.py (hunk 4 above)
Comment (PaulRC-ioet) [reply time_tracker_api/time_entries/time_entries_dao.py]: Done

**Review by enriquezrene — APPROVED** [review]

**Pull request merged**